File: CodingInterviewsPythonSolutions/739. Daily Temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        size = len(temperatures)
        result = [0] * size
        stk = []
        
        for i in range(size):
            while stk and temperatures[i] > temperatures[stk[-1]]:
                index = stk.pop()
                result[index] = i - index
            stk.append(i)
        
        return result

File: CodingInterviewsJavaSolutions/739. Daily Temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int size = temperatures.length;
        int[] result = new int[size];
        Deque<Integer> stack = new ArrayDeque<>();
        
        for (int i = 0; i < size; i++) {
            while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peekLast()]) {
                int index = stack.removeLast();
                result[index] = i - index;
            }
            stack.addLast(i);
        }
        
        return result;
    }
}
